sycl/doc/extensions/experimental/sycl_ext_oneapi_properties.asciidoc

```diff
@@ -80,9 +80,61 @@ properties P1{bar};
 properties P2{bar, foo{argv[0]}};
 properties P3{bar, baz};
 properties P4{baz, bar};
+
 static_assert(!std::is_same_v<decltype(P1), decltype(P2)>); // P1 and P2 contain different properties
 static_assert(!std::is_same_v<decltype(P1), decltype(P3)>); // P1 and P3 contain different properties
 static_assert(std::is_same_v<decltype(P3), decltype(P4)>); // The order properties are specified doesn't change the type
+
+```
+
+The alias `sycl::ext::oneapi::experimental::properties_t` is defined for
+convenience and it is defined as follows:
+
+```c++
+namespace sycl::ext::oneapi::experimental {
+
+template <auto... Props> using properties_t = decltype(properties{Props...});
+
+} // namespace experimental::oneapi::ext::sycl
+```
+
+Use this alias in situations where you need to get the type of a `properties`
+object that consists of 1 or more compile time constant properties. For example
+consider the following toy examples:
+
+.Examples using decltype
+```c++
+
+// buffer_location and awidth are some compile time constant properties
+annotated_arg<int *, decltype(properties{buffer_location<1>, awidth<32>})> a;
+
+// protocol_avalon_mm and uses_ready<false> are some compile time constant
+// properties
+using Pipe = pipe<..., decltype(properties{protocol_avalon_mm,
+                                           uses_ready<false>})>;
+
+// device_image_scope and host_access_read_write are some compile time constant
+// properties
+device_global<int, decltype(properties{device_image_scope,
+                                       host_access_read_write})> global_obj;
+```
```

Comment:
The rest of this extension uses abstract property names like "foo" and "bar".  I think these examples should follow suit to keep a consistent style.

I also think these examples give too much weight to the `decltype` approach.  Presumably, we want people to use `properties_t` instead, right?  To emphasize this, I'd word the description like:

> The type of the property list can be written with `properties_t` or with `decltype`, however the use of `properties_t` is preferred because it is less verbose. 
>
> [Example using `properties_t`]

I don't think you need any example showing `decltype`.  Why show examples for a style we don't recommend?

Reply:
That's a good point. I wanted to highlight the benefit of the approach. I agree that we should remove the decltype versions if we approve this PR.

Reply:
I think we do need to show the `decltype` version for users wanting to use C++17.

Reply:
Yes, I agree.  I wrote my comment above before I realized that the new mechanism requires C++20.

Reply:
Updated to leave the other parts untouched, just added a section about `properties_t` and called out the C++20 requirement.